data_validation/__main__.py

```diff
@@ -175,24 +187,42 @@ def build_config_from_args(args, config_manager):
         target_query_str = config_manager.get_query_from_file(args.target_query_file)
         config_manager.append_target_query(target_query_str)
 
+        # For custom-query column command
         if args.custom_query_type == consts.COLUMN_VALIDATION.lower():
             config_manager.append_aggregates(get_aggregate_config(args, config_manager))
 
-        elif not args.primary_keys:
-            raise ValueError(
-                "Expected valid primary keys for custom query row validation, got None."
+        # For custom-query row command
+        if args.custom_query_type == consts.ROW_VALIDATION.lower():
+
+            # Append calculated fields: --hash/--concat
+            config_manager.append_calculated_fields(
+                get_calculated_config(args, config_manager)
             )
 
-    config_manager.append_calculated_fields(get_calculated_config(args, config_manager))
+            # Append primary_keys
+            primary_keys = cli_tools.get_arg_list(args.primary_keys)
+            config_manager.append_primary_keys(
+                config_manager.build_column_configs(primary_keys)
+            )
 
+    # Append COLUMN_VALIDATION configs
     if config_manager.validation_type == consts.COLUMN_VALIDATION:
         config_manager.append_aggregates(get_aggregate_config(args, config_manager))
         if args.grouped_columns is not None:
             grouped_columns = cli_tools.get_arg_list(args.grouped_columns)
             config_manager.append_query_groups(
                 config_manager.build_column_configs(grouped_columns)
             )
-    elif config_manager.validation_type == consts.ROW_VALIDATION:
+
+    # Append ROW_VALIDATION configs
+    if config_manager.validation_type == consts.ROW_VALIDATION:
+
+        # Append calculated fields: --hash/--concat
+        config_manager.append_calculated_fields(
```

Comment:
See above comment - we should keep the append_calculated_fields outside the conditional

Reply:
Since `--hash` and `--concat` args are not present for `validate custom-query column` and `validate column`, it would throw an error while checking for `--hash` and ---concat`
https://github.com/GoogleCloudPlatform/professional-services-data-validator/blob/issue673-optimize-cli-tools-arg-parser/data_validation/__main__.py#L127